rct229/rules/section22/section22rule10.py

```diff
@@ -0,0 +1,148 @@
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        baseline_system_types_dict = get_baseline_system_types(rmi_b)
+        # create a list containing all HVAC systems that are modeled in the rmi_b
+        available_type_list = [
+            hvac_type
+            for hvac_type in baseline_system_types_dict.keys()
+            if len(baseline_system_types_dict[hvac_type]) > 0
+        ]
+
+        primary_secondary_loop_dict = get_primary_secondary_loops_dict(rmi_b)
+
+        return (
+            any(
+                [
+                    available_type in APPLICABLE_SYS_TYPES
+                    for available_type in available_type_list
+                ]
+            )
+            and len(primary_secondary_loop_dict) > 0
+        )
+
+    def create_data(self, context, data):
+        rmi_b = context.baseline
+
+        loop_pump_dictionary = {
+            pump["loop_or_piping"]: pump for pump in find_all("$.pumps[*]", rmi_b)
+        }
+
+        chw_loop_capacity_dict = {}
+        for chiller in find_all("$.chillers[*]", rmi_b):
+            if chiller["cooling_loop"] not in chw_loop_capacity_dict.keys():
+                chw_loop_capacity_dict[chiller["cooling_loop"]] = ZERO.POWER
+            chw_loop_capacity_dict[chiller["cooling_loop"]] += getattr_(
+                chiller, "chiller", "rated_capacity"
+            )
+
+        primary_secondary_loop_dict = get_primary_secondary_loops_dict(rmi_b)
+
+        return {
+            "loop_pump_dictionary": loop_pump_dictionary,
+            "chw_loop_capacity_dict": chw_loop_capacity_dict,
+            "primary_secondary_loop_dict": primary_secondary_loop_dict,
+        }
+
+    def list_filter(self, context_item, data):
+        fluid_loop_b = context_item.baseline
+        chw_loop_capacity_dict = data["chw_loop_capacity_dict"]
+
+        return fluid_loop_b["id"] in chw_loop_capacity_dict.keys()
+
+    class PrimaryFluidLoopRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section22Rule10.PrimaryFluidLoopRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section22Rule10.PrimaryFluidLoopRule.SecondaryChildLoopRule(),
+                index_rmr="baseline",
+                list_path="$.child_loops[*]",
+            )
+
+        def create_data(self, context, data):
+            fluid_loop_b = context.baseline
+            chw_loop_capacity_dict = data["chw_loop_capacity_dict"]
+
+            if (
+                chw_loop_capacity_dict[fluid_loop_b["id"]]
+                < MAX_FIXED_SPEED_CHW_LOOP_COOLING_CAPACITY
+            ):
+                target_secondary_pump_type = PUMP_SPEED_CONTROL.FIXED_SPEED
+            else:
+                target_secondary_pump_type = PUMP_SPEED_CONTROL.VARIABLE_SPEED
+
+            return {**data, "target_secondary_pump_type": target_secondary_pump_type}
```

Comment:
We used to need to include the `**data` part, but at some point I modified line 154 of `rule_list_base.py` to automatically include it.  Now you only need to return a dictionary of the pieces of data you want to add and it will be merged into the existing data structure.

Reply:
I didn't know about this. Thanks for letting me know about this. Fixed.